Data Race in go

```
=== RUN   TestContainerAttach
==================
WARNING: DATA RACE
Write at 0x00c4202fd060 by goroutine 74:
  os.(*file).close()
      /usr/local/go/src/os/file_unix.go:143 +0x10a
  os.(*File).Close()
      /usr/local/go/src/os/file_unix.go:132 +0x55
  github.com/containerd/containerd/vendor/github.com/containerd/fifo.(*fifo).Close.func1()
      /go/src/github.com/containerd/containerd/vendor/github.com/containerd/fifo/fifo.go:179 +0x69
  sync.(*Once).Do()
      /usr/local/go/src/sync/once.go:44 +0xe1
  github.com/containerd/containerd/vendor/github.com/containerd/fifo.(*fifo).Close()
      /go/src/github.com/containerd/containerd/vendor/github.com/containerd/fifo/fifo.go:182 +0x197
  github.com/containerd/containerd.(*wgCloser).Close()
      /go/src/github.com/containerd/containerd/io.go:136 +0x85
  github.com/containerd/containerd.(*IO).Close()
      /go/src/github.com/containerd/containerd/io.go:38 +0x76
  github.com/containerd/containerd.TestContainerAttach()
      /go/src/github.com/containerd/containerd/container_test.go:564 +0x18cd
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107

Previous read at 0x00c4202fd060 by goroutine 92:
  os.(*File).Read()
      /usr/local/go/src/os/file.go:98 +0x70
  github.com/containerd/containerd/vendor/github.com/containerd/fifo.(*fifo).Read()
      /go/src/github.com/containerd/containerd/vendor/github.com/containerd/fifo/fifo.go:138 +0x2fa
  io.copyBuffer()
      /usr/local/go/src/io/io.go:390 +0x116
  io.Copy()
      /usr/local/go/src/io/io.go:360 +0x7e
  github.com/containerd/containerd.copyIO.func3()
      /go/src/github.com/containerd/containerd/io_unix.go:66 +0xa7

Goroutine 74 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:697 +0x543
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:882 +0xaa
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107
  testing.runTests()
      /usr/local/go/src/testing/testing.go:888 +0x4e0
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:822 +0x1c3
  github.com/containerd/containerd.TestMain()
      /go/src/github.com/containerd/containerd/client_test.go:98 +0x31f
  main.main()
      github.com/containerd/containerd/_test/_testmain.go:84 +0x20f

Goroutine 92 (running) created at:
  github.com/containerd/containerd.copyIO()
      /go/src/github.com/containerd/containerd/io_unix.go:69 +0x595
  github.com/containerd/containerd.NewIOWithTerminal.func1()
      /go/src/github.com/containerd/containerd/io.go:66 +0x326
  github.com/containerd/containerd.(*container).NewTask()
      /go/src/github.com/containerd/containerd/container.go:170 +0xfa
  github.com/containerd/containerd.TestContainerAttach()
      /go/src/github.com/containerd/containerd/container_test.go:510 +0xc57
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107
==================
--- FAIL: TestContainerAttach (0.24s)
	testing.go:610: race detected during execution of test
```

I've seen this one only once but logging it there.

I'd say it's a go issue though.

/cc @stevvooe

Just saw in #1366

Included stack trace since it looks like may be updated line numbers
```
==================
WARNING: DATA RACE
Write at 0x00c42018e2e0 by goroutine 8:
  os.(*file).close()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/os/file_unix.go:143 +0x10a
  os.(*File).Close()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/os/file_unix.go:132 +0x55
  github.com/containerd/containerd/vendor/github.com/containerd/fifo.(*fifo).Close.func1()
      /home/travis/gopath/src/github.com/containerd/containerd/vendor/github.com/containerd/fifo/fifo.go:179 +0x69
  sync.(*Once).Do()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/sync/once.go:44 +0xe1
  github.com/containerd/containerd/vendor/github.com/containerd/fifo.(*fifo).Close()
      /home/travis/gopath/src/github.com/containerd/containerd/vendor/github.com/containerd/fifo/fifo.go:182 +0x197
  github.com/containerd/containerd.(*wgCloser).Close()
      /home/travis/gopath/src/github.com/containerd/containerd/io.go:183 +0x85
  github.com/containerd/containerd.(*cio).Close()
      /home/travis/gopath/src/github.com/containerd/containerd/io.go:65 +0x76
  github.com/containerd/containerd.TestContainerAttach()
      /home/travis/gopath/src/github.com/containerd/containerd/container_test.go:624 +0x18ae
  testing.tRunner()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:657 +0x107
Previous read at 0x00c42018e2e0 by goroutine 181:
  os.(*File).Read()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/os/file.go:98 +0x70
  github.com/containerd/containerd/vendor/github.com/containerd/fifo.(*fifo).Read()
      /home/travis/gopath/src/github.com/containerd/containerd/vendor/github.com/containerd/fifo/fifo.go:143 +0x24f
  io/ioutil.devNull.ReadFrom()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/io/ioutil/ioutil.go:144 +0xa3
  io/ioutil.(*devNull).ReadFrom()
      <autogenerated>:6 +0x7c
  io.copyBuffer()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/io/io.go:384 +0x35a
  io.Copy()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/io/io.go:360 +0x7e
  github.com/containerd/containerd.copyIO.func4()
      /home/travis/gopath/src/github.com/containerd/containerd/io_unix.go:79 +0xa7
Goroutine 8 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:697 +0x543
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:882 +0xaa
  testing.tRunner()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:657 +0x107
  testing.runTests()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:888 +0x4e0
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:822 +0x1c3
  github.com/containerd/containerd.TestMain()
      /home/travis/gopath/src/github.com/containerd/containerd/client_test.go:120 +0x91c
  main.main()
      github.com/containerd/containerd/_test/_testmain.go:108 +0x20f
Goroutine 181 (running) created at:
  github.com/containerd/containerd.copyIO()
      /home/travis/gopath/src/github.com/containerd/containerd/io_unix.go:82 +0x9bb
  github.com/containerd/containerd.NewIOWithTerminal.func1()
      /home/travis/gopath/src/github.com/containerd/containerd/io.go:103 +0x3c3
  github.com/containerd/containerd.(*container).NewTask()
      /home/travis/gopath/src/github.com/containerd/containerd/container.go:164 +0xfa
  github.com/containerd/containerd.TestContainerAttach()
      /home/travis/gopath/src/github.com/containerd/containerd/container_test.go:570 +0xc07
  testing.tRunner()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:657 +0x107
==================
--- FAIL: TestContainerAttach (0.25s)
	testing.go:610: race detected during execution of test
```

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -11,9 +11,15 @@ branches:
 environment:
   GOPATH: C:\gopath
   CGO_ENABLED: 1
+  GO_VERSION: 1.9
 
 before_build:
   - choco install -y mingw
+  # Install Go
+  - rd C:\Go /s /q
+  - appveyor DownloadFile https://storage.googleapis.com/golang/go%GO_VERSION%.windows-amd64.zip
+  - 7z x go%GO_VERSION%.windows-amd64.zip -oC:\ >nul
+  - go version
   # TODO: re-enable once the content unit-test have been updated to pass on windows
   #- choco install codecov
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,7 @@ services:
 language: go
 
 go:
-  - 1.8.x
+  - 1.9.x
 
 go_import_path: github.com/containerd/containerd
 

